Update SES Version - Vulnerability Fix

andrewpeters9 opened the pull request:
## Explanation

* Vulnerability has been found recently with `SES<0.16.0` - https://github.com/advisories/GHSA-whpx-q3rq-w8jc
*  Vulnerable versions of the packages are currently loaded in via 
  a) directly via the project's workspace and 
  b) via `@metamask/phishing-warning`
* Package changelog: https://github.com/endojs/endo/blob/master/packages/ses/CHANGELOG.md

## Manual Testing Steps
* `yarn && yarn start`
* Open a URL within the `@metamask/phishing-warning` blocked-list to ensure that the plugin is working as intended
  * Example: https://thekrakenz.com/
* Open the extension in tab form, inspect to see that `lockdown-install.js` is still loaded - i.e., the SES code is still loaded
  * https://gyazo.com/9de8e66a6de39f1a6d9bef02f6b336c7

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue

andrewpeters9 committed:
package bumped

andrewpeters9 committed:
update ses import dir

andrewpeters9 committed:
agoric references removed

andrewpeters9 committed:
domainTaming set to unsafe

andrewpeters9 commented on app/scripts/lockdown-run.js:
```diff
@@ -6,6 +6,7 @@ try {
     errorTaming: 'unsafe',
     mathTaming: 'unsafe',
     dateTaming: 'unsafe',
+    domainTaming: 'unsafe',
```
Added due to this breaking change:

https://github.com/endojs/endo/blob/master/packages/ses/CHANGELOG.md#0150-2021-11-02

andrewpeters9 commented:
I have read the CLA Document and I hereby sign the CLA

andrewpeters9 commented on test/helpers/protect-intrinsics-helpers.js:
```diff
@@ -13,14 +13,10 @@ module.exports = {
   const ignoreList = new Set([
     'Compartment',
-    'HandledPromise',
-    'StaticModuleRecord',
```
`@agoric/*` packages no longer pulled in due to the version change

Gudahtt commented on package.json:
```diff
@@ -91,6 +91,7 @@
     "test-storybook:ci": "concurrently -k -s first -n \"SB,TEST\" -c \"magenta,blue\" \"yarn storybook:build && npx http-server storybook-build --port 6006 \" \"wait-on tcp:6006 && yarn test-storybook --maxWorkers=2\""
   },
   "resolutions": {
+    "@metamask/phishing-warning/ses": "^0.18.1",
```
v1.2.2 of the phishing warning page has just been published - you can update that here to point the extension at the new page, and drop this resolution

Gudahtt reviewed (dismissed):
LGTM!

Though we'll want to flag this to QA as needing thorough regression testing

andrewpeters9 commented:
> LGTM!
> 
> Though we'll want to flag this to QA as needing thorough regression testing

Thanks. I've added more testing info and I've added the QA label. Is there anything else I need to do? @Gudahtt

Gudahtt commented:
Thanks! No further action needed as far as I'm aware.

andrewpeters9 commented:
@Gudahtt Am I free to merge this directly after two approvals? Or does the QA team need to test it on the branch before it gets merged?

legobeat committed:
Merge branch 'develop' into update-ses

andrewpeters9 committed:
Merge branch 'develop' into update-ses
